packages/manager/test/SliceMachineManager-plugin-dangerouslyCallHook.test.ts

```diff
@@ -0,0 +1,31 @@
+import { expect, it } from "vitest";
+
+import { createTestPlugin } from "./__testutils__/createTestPlugin";
+import { createTestProject } from "./__testutils__/createTestProject";
+
+import { createSliceMachineManager } from "../src";
+
+it("calls a given plugin hook", async () => {
+	const adapter = createTestPlugin({
+		setup: ({ hook }) => {
+			hook("debug", (data) => data);
+		},
+	});
+	const cwd = await createTestProject({ adapter });
+	const manager = createSliceMachineManager({
+		nativePlugins: { [adapter.meta.name]: adapter },
+		cwd,
+	});
+
+	await manager.plugins.initPlugins();
+
+	const hookReturnValue = await manager.plugins.dangerouslyCallHook(
```

Comment:
Calling hooks directly through the plugin runner is dangerous because we don't have any guards are runtime type checking around them. The dedicated methods that use plugin hooks (e.g. `slices.readSlice()`) use codecs to safely return data and reshape data to be more convenient to use.

Never use `dangerouslyCallHook()` unless you know what you are doing. Prefer using or writing safe manager methods instead. 🙂